examples/buffer/src/main.rs

```diff
@@ -1,15 +1,13 @@
 use bindings::{Windows::Foundation::*, Windows::Win32::System::WinRT::IMemoryBufferByteAccess};
 use windows::*;
 
-fn as_slice(buffer: &IMemoryBufferReference) -> Result<&mut [u8]> {
+unsafe fn as_slice(buffer: &IMemoryBufferReference) -> Result<&mut [u8]> {
```

Comment:
Could perhaps use `/// # Safety` documentation, explaining that `IMemoryBufferReference` performs no borrow checking (by means of refcounting) and allows multiple mutators accessing the data at the same time (which is forbidden by `&mut`, which pretty much means having unique ownership of the data
).

Reply:
In addition, should this function be called `as_mut_slice` to match similar functions in `std` more closely?